### PR TITLE
Add IPVS netlink support

### PR DESCRIPTION
**Pull request by mrjana**
This PR adds netlink support to manipulate ipvs configuration.

Signed-off-by: Jana Radhakrishnan <mrjana@docker.com>

### CODE REVIEW

File: ipvs/netlink.go [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+	return 0, fmt.Errorf("no family id in the netlink response")
+}
+
+func rawIPData(ip net.IP) []byte {
+	family := nl.GetIPFamily(ip)
+	if family == nl.FAMILY_V4 {
+		return ip.To4()
+	}
+
+	return ip.To16()
```
Comment (aboch) [inline ipvs/netlink.go]: The `.To16()` is not needed.
If family is `nl.FAMILY_V6`, then it is already an IPv6 address, an array of 16 bytes.

**Comment by aboch** [comment]
Besides a minor comment, LGTM

**Comment by mavenugo** [comment]
LGTM

**Comment by mqliang** [comment]
@mrjana First, I am sorry this is not an issue, but a question. I want to implement a "ipvsadm --restore" like function. I first implement like this:


```
type ServiceDestination struct {
	Service *Service
	Destinations []*Destination
}

func ( i *IPVSHandle) Restore(items []ServiceDestination)([][]byte, error) {
	req := nl.NewNetlinkRequest(i.ipvsFamily, syscall.NLM_F_REPLACE)
	req.AddData(&genlMsgHdr{cmd: ipvsCmdSetConfig, version: 1})


	listAttr := nl.NewRtAttr(ipvsCmdAttrUnspec, nl.Uint16Attr(uint16(i.ipvsFamily)))

	for _, item := range  items {
		itemAttr := nl.NewRtAttr(ipvsCmdAttrUnspec, nl.Uint16Attr(uint16(i.ipvsFamily)))

		srvAttr := toServiceAttr(item.Service)
		nl.NewRtAttrChild(itemAttr, ipvsCmdAttrService, srvAttr.Serialize())

		for _, d := range item.Destinations {
			nl.NewRtAttrChild(itemAttr, ipvsCmdAttrDest, fillDestinaton(d).Serialize())
		}

		nl.NewRtAttrChild(listAttr, ipvsCmdAttrUnspec, itemAttr.Serialize())
	}

	req.AddData(listAttr)
	return req.Execute(syscall.NETLINK_GENERIC, 0)
}


func toServiceAttr(s *Service) *nl.RtAttr {
	cmdAttr := nl.NewRtAttr(ipvsCmdAttrService, nil)
	nl.NewRtAttrChild(cmdAttr, ipvsSvcAttrAddressFamily, nl.Uint16Attr(s.AddressFamily))
	if s.FWMark != 0 {
		nl.NewRtAttrChild(cmdAttr, ipvsSvcAttrFWMark, nl.Uint32Attr(s.FWMark))
	} else {
		nl.NewRtAttrChild(cmdAttr, ipvsSvcAttrProtocol, nl.Uint16Attr(s.Protocol))
		nl.NewRtAttrChild(cmdAttr, ipvsSvcAttrAddress, rawIPData(s.Address))

		// Port needs to be in network byte order.
		portBuf := new(bytes.Buffer)
		binary.Write(portBuf, binary.BigEndian, s.Port)
		nl.NewRtAttrChild(cmdAttr, ipvsSvcAttrPort, portBuf.Bytes())
	}

	nl.NewRtAttrChild(cmdAttr, ipvsSvcAttrSchedName, nl.ZeroTerminated(s.SchedName))
	if s.PEName != "" {
		nl.NewRtAttrChild(cmdAttr, ipvsSvcAttrPEName, nl.ZeroTerminated(s.PEName))
	}

	f := &ipvsFlags{
		flags: s.Flags,
		mask:  0xFFFFFFFF,
	}
	nl.NewRtAttrChild(cmdAttr, ipvsSvcAttrFlags, f.Serialize())
	nl.NewRtAttrChild(cmdAttr, ipvsSvcAttrTimeout, nl.Uint32Attr(s.Timeout))
	nl.NewRtAttrChild(cmdAttr, ipvsSvcAttrNetmask, nl.Uint32Attr(s.Netmask))
	return cmdAttr
}
```

But it does not work, could please kindly point out what's going wrong.

**Pull request merged**